### PR TITLE
[KEH-352] - Add MkDocs to project to reference API 

### PR DESCRIPTION
**Pull request by evisos-ons**
Run `make install-docs` and then `make mkdocs` to run the MkDocs. Make sure nothing is running on localhost:8000.

### CODE REVIEW

File: .github/workflows/deploy_mkdocs.yml
Comment (TotalDwarf03) [inline .github/workflows/deploy_mkdocs.yml:24]: I believe this action will fail because it's pushing straight to main which violates org rules. 

Happy to leave for now and find out if it works. This is the action used in other repos: https://github.com/ONS-Innovation/github-copilot-usage-dashboard/blob/master/.github/workflows/deploy_mkdocs.yml

I know it's not as clean as `mkdocs gh-deploy` as it relies on a manual build, but it's a work around for the org rules.

File: docs/api.md
Comment (TotalDwarf03) [inline docs/api.md:7]: Button below this iframe to go to the url would be nice.

File: docs/assets/creation_flow.png
Comment (TotalDwarf03) [inline docs/assets/creation_flow.png:1]: Is this process still the same after the terraform changes?

File: docs/deployment.md
Comment (TotalDwarf03) [inline docs/deployment.md:23]: Link to confluence page for versioning here perhaps?

File: docs/deployment.md
Comment (TotalDwarf03) [inline docs/deployment.md:25]: More of an FYI, you can actually get the push commands from the ERC repository. There is a view push commands button. All you have to do is then change the versioning of the tag

File: docs/infrastructure.md
Comment (TotalDwarf03) [inline docs/infrastructure.md:1]: Just to confirm if this process is still the same after the terraform changes?

File: docs/infrastructure.md
Comment (TotalDwarf03) [inline docs/infrastructure.md:43]: Unfinished sentence

File: docs/infrastructure.md
Comment (TotalDwarf03) [inline docs/infrastructure.md:39]: Does `aws_cognito_token_url` ever get set?

**Review by TotalDwarf03 — COMMENTED** [review]
Few comments. Going to try and run the docs locally to proof everything. Happy to approve once comments are address and working locally.

File: docs/deployment.md
Comment (evisos-ons) [reply docs/deployment.md:25]: Should I remove this from the docs then?

Comment (TotalDwarf03) [reply docs/deployment.md:25]: No it's fine to include some commands 👍

File: docs/assets/creation_flow.png
Comment (evisos-ons) [reply docs/assets/creation_flow.png:1]: Yes, updated on Confluence as well.

File: docs/deployment.md
Comment (evisos-ons) [reply docs/deployment.md:23]: To the KEH? If so, I have now added.

File: docs/infrastructure.md
Comment (evisos-ons) [reply docs/infrastructure.md:1]: Yes, updated on Confluence as well.

File: docs/infrastructure.md
Comment (evisos-ons) [reply docs/infrastructure.md]: Changed.

File: docs/infrastructure.md
Comment (evisos-ons) [reply docs/infrastructure.md]: No, I made changes here but they didn't commit for some reason. Updated now.

File: docs/deployment.md
Comment (evisos-ons) [reply docs/deployment.md:27]: Alright, thanks 🙏

File: .github/workflows/deploy_mkdocs.yml
Comment (TotalDwarf03) [inline .github/workflows/deploy_mkdocs.yml:40]: Need to make this directory using `mkdocs build` then rename the `/site` directory to `mkdocs_deployment`.

This is also worth adding to your readme.

**Review by TotalDwarf03 — APPROVED** [review]
Runs fine locally and reads well. I've left a comment for you to deploy the documentation using the GitHub action. Going to approve on the basis that you'll do this **before merging**.

File: .github/workflows/deploy_mkdocs.yml
Comment (evisos-ons) [reply .github/workflows/deploy_mkdocs.yml:40]: Done 👍

**Pull request merged**
